torch/_refs/__init__.py

```diff
@@ -577,29 +577,32 @@ def logsumexp(
 def nan_to_num(
     a: TensorLikeType,
     *,
-    nan: Optional[NumberType] = 0.0,
-    posinf: Optional[NumberType] = None,
-    neginf: Optional[NumberType] = None,
+    nan: Optional[float] = 0.0,
```

Comment:
These are coded as floats in native_functions.yaml but Python but letting me numbers also seems fine (and it would support complex)